pkgs/development/python-modules/catboost/default.nix

```diff
@@ -0,0 +1,51 @@
+{ lib, buildPythonPackage, clang_7, fetchFromGitHub, graphviz, matplotlib, numpy
+, pandas, plotly, python2, python3, pythonOlder, scipy, six, withCuda ? false
+, cudatoolkit }:
+
+buildPythonPackage {
+  pname = "catboost";
+  version = "0.24.4";
+
+  disabled = pythonOlder "3.4";
+
+  src = fetchFromGitHub {
+    owner = "catboost";
+    repo = "catboost";
+    rev = "2cb3eda7dd228e851d9817eb018b9d7521ac342d";
+    sha256 = "sha256-pzmwEiKziB4ldnKgeCsP2HdnisX8sOkLssAzNfcSEx8=";
+  };
+
+  # Passed to make in setup.py.
+  nativeBuildInputs = [ clang_7 python2 ];
+
+  propagatedBuildInputs = [ graphviz matplotlib numpy pandas scipy plotly six ]
+    ++ lib.optional withCuda [ cudatoolkit ];
+
+  # First apply patches, then go to source root; sourceRoot applies too early.
+  patches = [ ./nix-support.patch ./format.patch ];
```

Comment:
```suggestion
  patches = [
    ./nix-support.patch
    ./format.patch
  ];
```

Reply:
Re: all formatting suggestions. I assumed nixfmt would yield a satisfying result. Is there a style guide? I'll follow your suggestions anyway.

Reply:
Not really a comprehensive one.